client/src/components/ChannelFilter.tsx

```diff
@@ -0,0 +1,64 @@
+import Select, {SingleValue} from 'react-select';
+import React from 'react';
+import styled from 'styled-components';
+import {useCallback, useEffect, useState} from 'react';
+import {fetchData} from '../services/fetch';
+
+const FilterDiv = styled.div`
+      min-width: 350px;
+      max-width: 50%;
+      flex-grow: 1;
+    `;
+
+const ChannelFilter = ({onSelected, selectedChannel}: { onSelected: CallableFunction; selectedChannel: string; }) => {
+    const [channelOptions, setChannelOptions] = useState([] as { label: string; value: string; }[]);
+    const [selectPlaceHolder, setSelectPlaceHolder] = useState('Code Reviews For All Slack Channels');
+    const handleChannelSelectionChange = useCallback((data: SingleValue<{ label: string; value: string; }>) => {
```

Comment:
This is a callback function to trigger the onSelected even when Select component trigger the value updated.  It's bubble up the the parent component via the onSelected callback.   The useCallback is to create a reference to avoid side potential side affect where a function is declared, it will always have a new reference.  useCallback would keep that reference the same if dependencies have not changed.

Reply:
I've refactored the code to be more efficient and remove un-needed code in commit d7056ba4b2bb4e63b27c4167d15e3a659100ee54